plasmapy/particles/tests/test_atomic.py

```diff
@@ -43,345 +43,208 @@
-    ["He 1+", {"mass_numb": 99}, InvalidParticleError],
-    ["fe-56 1+", {}, InvalidParticleError],
-    ["H-1", {"mass_numb": 1, "Z": 1}, ParticleWarning],
-    ["H", standard_atomic_weight("H")],
-]
-
-is_stable_table = [
-    ["H-1", True],
-    [(1, 1), True],
-    ["N-14", True],
-    [("N", 14), True],
-    ["P-31", True],
-    [("P", 31), True],
-    ["p", True],
-    ["alpha", True],
-    ["Xe-124", True],
-    ["Fe", {"mass_numb": 56}, True],
-    ["Fe-56", True],
-    ["iron-56", True],
-    ["Iron-56", True],
-    [(26, 56), True],
-    ["Be-8", False],
-    ["U-235", False],
-    ["uranium-235", False],
-    ["T", False],
-    [(4, 8), False],
-    ["tritium", False],
-    ["Pb-209", False],
-    ["lead-209", False],
-    ["Lead-209", False],
-    ["Pb", {"mass_numb": 209}, False],
-    [(82, 209), False],
-    [("hydrogen-444444",), InvalidParticleError],
-    [("hydrogen", 0), InvalidParticleError],
-    [("",), InvalidParticleError],
-    [("pb-209",), InvalidParticleError],
-    [("h",), InvalidParticleError],
-    [("He",), InvalidIsotopeError],
-    [("B",), InvalidIsotopeError],
-]
 
-integer_charge_table = [
-    ["H+", 1],
-    ["D +1", 1],
-    ["tritium 1+", 1],
-    ["H-", -1],
-    ["Fe -2", -2],
-    ["Fe 2-", -2],
-    ["N--", -2],
-    ["N++", 2],
-    ["alpha", 2],
-    ["proton", 1],
-    ["deuteron", 1],
-    ["triton", 1],
-    ["electron", -1],
-    ["e-", -1],
-    ["e+", 1],
-    ["positron", 1],
-    ["n", 0],
-    ["neutron", 0],
-    ["p-", -1],
-    ["antiproton", -1],
-    ["fads", InvalidParticleError],
-    ["H++", InvalidParticleError],
-    ["h+", InvalidParticleError],
-    ["fe 1+", InvalidParticleError],
-    ["d+", InvalidParticleError],
-    ["Fe 29+", InvalidParticleError],
-    ["H-1", ChargeError],
-    ["H---", ParticleWarning],
-    ["Fe -26", ParticleWarning],
-    ["Og 10-", ParticleWarning],
-]
-
-electric_charge_table = [
-    ["p", u.C],
-    ["p", 1.6021766208e-19 * u.C],
-    ["e", -1.6021766208e-19 * u.C],
-    ["alpha", 3.2043532416e-19 * u.C],
-    ["n", 0 * u.C],
-    ["badinput", InvalidParticleError],
-    ["h+", InvalidParticleError],
-    ["Au 81+", InvalidParticleError],
-    ["Au 81-", ParticleWarning],
-    ["H---", ParticleWarning],
-]
-
-half_life_table = [["H-1", u.s], ["tritium", u.s], ["H-1", np.inf * u.s]]
+@pytest.mark.parametrize(
+    "tested_function, args, kwargs, expected_output",
+    table_functions_args_kwargs_output,
+)
+def test_functions_and_values(tested_function, args, kwargs, expected_output):
+    actual_output = tested_function(*args, **kwargs)
+    if isinstance(expected_output, u.Quantity):
+        assert_quantity_allclose(actual_output, expected_output)
+    elif isinstance(expected_output, u.UnitBase):
+        assert actual_output.unit == expected_output
+    elif isinstance(expected_output, (int, float, str)):
+        assert expected_output == actual_output
+    else:
+        pytest.fail("Other types!")
```

Comment:
Since code similar to this comes up in `particles`, `formulary`, `utils`, and probably future subpackages, it'd be helpful to perform a tradeoff analysis of the following possibilities:
 - Include this code (or similar) whenever we perform a test that compares two objects
 - Create a function that performs the comparison, and then use that function whenever this comes up

Reply:
That's a fair point, this is generalizable - but it's not something I'd prepare ahead of time. I would personally not write that code before I really start getting annoyed at how much I'm rewriting this kind of check.

Who knows, maybe by next month we'll write all our tests with pytest-regressions and hypothesis ;)

Reply:
> I would personally not write that code before I really start getting annoyed at how much I'm rewriting this kind of check.

This is pretty much exactly the story of why I ended up writing the test helpers in the first place! 😹

Reply:
I'll point out that `np.allclose` and `np.isclose` will compare units and values.  Well, yes and no.  If it detects the units are equivalent then it'll convert and check.  So, if you want strict unit comparison, then you'll have to explicitly check units.

My biggest concern with developing a single "comparison" function to use across all tests is that I have no confidence that we can achieve something that is detailed enough and generalized enough to work for all cases.  With almost 100% certainty we'll come across cases that the comparison function won't work for.  Additionally, it might make debugging more difficult since the comparisons are trapped inside a function and might muddle up the Traceback (I could be mistaken about this though.)

What I foresee happening if we write a "comparison" function like this is that when we run across a scenario where it doesn't work perfectly we have limit non-optimal solutions.  (1) We try to update the comparison function, but the updates might not work with other scenarios.  In this case we'd end up breaking tests throughout the code. (2) We try to shoehorn the comparison function into our test code and code around it to make it work.   **Now, I could be wrong about all of this and I don't think we could know for sure unless we try to implemented it.**  I just don't see it as our most fruitful endeavor at this point.

Reply:
> My biggest concern with developing a single "comparison" function to use across all tests is that I have no confidence that we can achieve something that is detailed enough and generalized enough to work for all cases. 

The purpose isn't to find a single comparison function that can do everything, but rather to find a comparison function that works for tests that come up over and over and over that fit the result test pattern.  

> Now, I could be wrong about all of this and I don't think we could know for sure unless we try to implemented it.

I've already implemented it!  #927 

> I just don't see it as our most fruitful endeavor at this point.

I'm planning to use the test helper functionality for as long as I'm working with scientific Python, so for my own projects, I want to actually be able to start using it.